aws_lambda_powertools/utilities/parser/envelopes/sqs.py

```diff
@@ -19,7 +19,7 @@ class SqsEnvelope(BaseEnvelope):
     all items in the list will be parsed as str and npt as JSON (and vice versa)
     """
 
-    def parse(self, data: Dict[str, Any], model: Union[BaseModel, str]) -> List[Union[BaseModel, str]]:
+    def parse(self, data: Dict[str, Any], model: Union[BaseModel, str]) -> List[BaseModel]:
```

Comment:
The abstract method doesn't accept `str` for the model arguments. 
I guess the model is not `BaseModel` `Type[BaseModel]`
Also, I must recommend using TypeVar for parse methods and _parse method.

example
```python
Model = TypeVar("Model", bound=BaseModel)
...
def parse(self, data: Dict[str, Any], model: Type[Model]) -> List[Model]:
```
I found a good example of this case in this repo.
https://github.com/heitorlessa/aws-lambda-powertools-python/blob/9e4286399527b32a0cc2fee44362ea4864e46383/aws_lambda_powertools/utilities/parser/parser.py#L89

Reply:
Ah yes! That's leftover from the refactor - Addressing this now